### PR TITLE
[Purrfect-UI] Tag

### PR DESCRIPTION
**Pull request by Rcaradec**

### CODE REVIEW

**Comment by MatouTechnologies** [comment]
This pull request has been deployed to Vercel.
<table>
  <tr>
    <td><strong>✅ Preview:</strong></td>
    <td><a href='https://matou-technologies-purrfect-ui-storybook-35.vercel.app'>https://matou-technologies-purrfect-ui-storybook-35.vercel.app</a></td>
  </tr>
  <tr>
    <td><strong>🔍 Inspect:</strong></td>
    <td><a href='https://vercel.com/matou-technologies/purrfect-ui/H3aZ2mYCd1s4bpi13nQGjETQegun'>https://vercel.com/matou-technologies/purrfect-ui/H3aZ2mYCd1s4bpi13nQGjETQegun</a></td>
  </tr>
</table>


<hidden purpose="for-rewritable-pr-comment-action-use" value="deploy-preview-storybook"></hidden>

File: src/display/Tag/Tag.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  return css`
+    color: ${theme.color.text[getContrastYIQ(color)]};
+    background-color: ${color};
+    &:hover:enabled {
+      opacity: 0.9;
+    }
+  `;
+};
+
+export type TagKind = "normal" | "outlined";
```
Comment (Balkeo) [inline src/display/Tag/Tag.tsx:38]: Is the export needed here ?

File: src/icons/delete.tsx [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import React from "react";
 
 type DeleteIconProps = {
-  onClick: () => void;
+  onClick?: () => void;
```
Comment (Balkeo) [inline src/icons/delete.tsx:4]: Why an onClick here ?

File: src/display/Tag/Tag.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  onPress?: () => void;
+}>`
+  ${body1}
+  ${({ size, kind, variant, onPress }) => css`
+    ${getTagSizeStyles(size)};
+    ${tagColor(variant)};
+    ${tagKind(kind, variant)};
+    ${onPress &&
+    `
+    cursor: pointer;
```
Comment (MarcoPOtard) [inline src/display/Tag/Tag.tsx:76]: Is the cursor pointer necessary ? It's already on the hover

File: src/icons/delete.tsx [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
+  style?: React.CSSProperties;
```
Comment (MarcoPOtard) [inline src/icons/delete.tsx:5]: Better than style, I think we have to create size props in the icon components like
`{ size = 20 }: { size?: number }`
and 
`     height={size}
      width={size}
`

File: src/display/Tag/Tag.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+export const WithIcon: Story = {
+  args: {
+    children: "Tag",
+    variant: "info",
+    size: "large",
+    trailingIcon: (
+      <DeleteIcon
+        onClick={() => alert("Do something with the tag")}
+        style={{ height: "14px", width: "14px" }}
```
Comment (MarcoPOtard) [inline src/display/Tag/Tag.stories.tsx:78]: switch to a size props, see the comment in delete component

File: src/display/Tag/Tag.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import Tag from "src/display/Tag/Tag";
+import DeleteIcon from "src/icons/delete";
+
+describe("Tag Component", () => {
+  it("renders the base tag", () => {
+    render(<Tag>Tag</Tag>);
+    expect(screen.getByText("Tag")).toBeInTheDocument();
+  });
+
+  it("renders the Tag with a variant prop", () => {
```
Comment (MarcoPOtard) [inline src/display/Tag/Tag.spec.tsx:13]: This test is the same the first one. I don't understand what is the différence

File: src/icons/delete.tsx (hunk 2 above)
Comment (Rcaradec) [reply src/icons/delete.tsx:4]: removed

File: src/icons/delete.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rcaradec) [reply src/icons/delete.tsx:5]: removed

File: src/display/Tag/Tag.tsx [hunk 7]
```diff
@@ -0,0 +1,139 @@
+import React from "react";
+import { useButton, usePress } from "react-aria";
+import styled, { css } from "styled-components";
+import { Variant } from "src/action/Button/Button";
+import { body1, theme } from "src/guidelines/theme";
+import { getContrastYIQ, sizing } from "src/utils/utils";
+
+export type TagSizeType = "normal" | "large";
```
[the remark this answers is not in the document]
Comment (Rcaradec) [reply src/display/Tag/Tag.tsx:8]: change to local declaration

File: src/display/Tag/Tag.tsx (hunk 3 above)
Comment (Rcaradec) [reply src/display/Tag/Tag.tsx:76]: removed

File: src/display/Tag/Tag.stories.tsx (hunk 5 above)
Comment (Rcaradec) [reply src/display/Tag/Tag.stories.tsx:78]: new Styled Icon behavior, icon size is handled in the component where it's used.

File: src/display/Tag/Tag.spec.tsx (hunk 6 above)
Comment (Rcaradec) [reply src/display/Tag/Tag.spec.tsx:13]: modification based on Balkeo comment : 
https://github.com/Groupe-Gaillard/purrfect-ui/pull/35#discussion_r1671790684

Thus, this test won't be needed anymore

**Comment by Rcaradec** [comment]
Tag List Story
image

File: src/display/Tag/Tag.stories.tsx
[the remark this answers is not in the document]
Comment (Rcaradec) [reply src/display/Tag/Tag.stories.tsx:1]: Tag List Story

image

File: src/display/Tag/Tag.spec.tsx [hunk 8]
```diff
@@ -45,8 +39,9 @@ describe("Tag Component", () => {
   });
 
   it("renders the Tag with an icon", () => {
-    render(<Tag leadingIcon={<DeleteIcon />}>With Icon</Tag>);
-    const icon = screen.getByTestId("delete-icon");
+    render(<Tag leadingIcon={<AddIcon />}>With Icon</Tag>);
+    // Adjust the query to match the role and accessible name of the button element
+    const icon = screen.getByRole("button", { name: "With Icon" });
```
Comment (MarcoPOtard) [inline src/display/Tag/Tag.spec.tsx:43]: Là, j'ai l'impression que tu ne testes pas la présence de l'icon

File: src/display/Tag/Tag.tsx [hunk 9]
```diff
@@ -3,20 +3,31 @@ import { useButton, usePress } from "react-aria";
...
       `;
     case "large":
       return css`
-        max-height: ${sizing(20)};
+        height: ${sizing(20)};
         padding: ${sizing(4)} ${sizing(8)};
+
+        & > ${IconSVG} {
+          height: ${sizing(20)};
+          width: ${sizing(16)};
```
Comment (MarcoPOtard) [inline src/display/Tag/Tag.tsx:29]: attention les 2 tailles sont censées être identiques ;)

**Review by MarcoPOtard — APPROVED** [review]

**Review by Balkeo — APPROVED** [review]

**Pull request merged**